File: src/python/finished/count_partitions_with_even_sum_difference.py
from typing import List

# https://leetcode.com/problems/count-partitions-with-even-sum-difference/


class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        total = sum(nums)
        count = 0
        cur = 0
        for n in range(len(nums) - 1):
            cur += n
            if (total - cur * 2) % 2 == 0:
                count += 1
        return count

File: src/rust/finished/count_partitions_with_even_sum_difference.rs
impl Solution {
    pub fn count_partitions(nums: Vec<i32>) -> i32 {
        let total: i32 = nums.iter().sum();
        let mut cur: i32 = 0;
        let mut count = 0;
        for n in nums.iter().take(nums.len() - 1) {
            cur += n;
            if (total - cur * 2).rem_euclid(2) == 0 {
                count += 1;
            }
        }
        count
    }
}
